File: Project 1 Question 4/src/Pachyderm.java
/**
 * 
 * This is a subclass of Animal Class for Pachyderm Animal Family
 *
 */
public class Pachyderm extends Animal{

	
	/**
	 * A constructor to initialize Pachyderm's name
	 * @param name Name of the Pachyderm
	 */
	public Pachyderm(String name)
	{
		super(name);
	}
	/**
	 * Method which tell Pachyderm to walk
	 */
	public void roam()
	{
		System.out.println(this.name+" is a "+this.getClass().getName() );
		System.out.println(this.name+" walked");
	}
	/**
	 * Method which tells Pachyderm to eat grass
	 */
	public void eat()
	{
		System.out.println(this.name+" is a "+this.getClass().getName() );
		System.out.println(this.name+" ate grass");
	}
	
	
}

File: Project 2/Pachyderm.cpp
#include "Pachyderm.h"

#include<iostream>
using namespace std;

Pachyderm ::Pachyderm(string name, string type):Animal(name,type)
{
eat_ = new Grass();
}

void Pachyderm::Roam()
{
  cout<<name_<<" is a "<<type_<<endl;
  cout<<name_<<" walked\n";
}
Hippo:: Hippo(string name):Pachyderm(name,"Hippo")
{

}

void Hippo::MakeNoise()
{
  cout<<name_<<" is a "<<type_<<endl;
  cout<<name_<<" growled\n";
}


Rhino::Rhino(string name):Pachyderm(name,"Rhino")
{
  
}

void Rhino::MakeNoise()
{
  cout<<name_<<" is a "<<type_<<endl;
  cout<<name_<<" bellowed\n";
}

Elephant::Elephant(string name):Pachyderm(name,"Elephant")
{
  
}

void Elephant::MakeNoise()
{
  cout<<name_<<" is a "<<type_<<endl;
  cout<<name_<<" trumptted\n";
}